ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/FibonacciReversalIndicator.java

```diff
@@ -35,62 +36,46 @@
  */
 public class FibonacciReversalIndicator extends RecursiveCachedIndicator<Decimal> {
 
-    private PivotPointIndicator pivotPointIndicator;
-    private PivotLevel level;
-
-    public enum PivotLevel{
-        RESISTANCE_3(Decimal.ONE, true),
-        RESISTANCE_2(Decimal.valueOf(0.618), true),
-        RESISTANCE_1(Decimal.valueOf(0.382), true),
-        SUPPORT_1(Decimal.valueOf(0.382), false),
-        SUPPORT_2(Decimal.valueOf(0.618), false),
-        SUPPORT_3(Decimal.ONE, false);
-
-        private Decimal fibFactor;
-        private boolean isResistance;
-
-        PivotLevel(Decimal fibFactor, boolean isResistance){
-            this.fibFactor = fibFactor;
-            this.isResistance = isResistance;
-        }
-
-        public Decimal getFactor(){
-            return this.fibFactor;
-        }
-
-        public boolean isResistance(){
-            return isResistance;
-        }
+    private final PivotPointIndicator pivotPointIndicator;
+    private final FibReversalTyp fibReversalTyp;
+    private final Decimal fibonacciFactor;
 
+    public enum FibReversalTyp{
+        SUPPORT,
+        RESISTANCE
     }
 
 
     /**Constructor.
      * <p>
-     * Calculates the (fibonacci) support 3 reversal
+     * Calculates a (fibonacci) reversal
      * @param pivotPointIndicator the {@link PivotPointIndicator} for this reversal
+     * @param fibonacciFactor the fibunacci factor for this reversal (e.g. 1, 0.618, 0.382, ...)
```

Comment:
add this within this class:
```

public enum FibonacciFactor {
		Factor1(Decimal.valueOf(0.382)), 
		Factor2(Decimal.valueOf(618)), 
		Factor3(Decimal.ONE);

		FibonacciFactor(Decimal factor) {
			this.factor = factor;
		}
		private final Decimal factor;
	}
```

and add a additonal constructor with FibonacciReversalIndicator(FibonacciFactor factor,..) for user convenience. The 3 factors above are the "standard fibonacci factors". Look at my source in https://github.com/ta4j/ta4j/pull/49/ where I add a additional constructor for this.